plugins/tekton/src/plugin.ts

```diff
@@ -4,12 +4,74 @@ import '@patternfly/patternfly/patternfly-charts-theme-dark.css';
 import '@patternfly/patternfly/utilities/Accessibility/accessibility.css';
 
 import {
+  createApiFactory,
   createComponentExtension,
   createPlugin,
+  discoveryApiRef,
+  fetchApiRef,
+  gitlabAuthApiRef,
+  googleAuthApiRef,
+  microsoftAuthApiRef,
+  oktaAuthApiRef,
+  oneloginAuthApiRef,
 } from '@backstage/core-plugin-api';
 
+import {
+  KubernetesAuthProviders,
+  KubernetesBackendClient,
+} from '@janus-idp/shared-react';
+
+import { kubernetesApiRef, kubernetesAuthProvidersApiRef } from './types/types';
+
 export const tektonPlugin = createPlugin({
   id: 'tekton',
+  apis: [
+    createApiFactory({
+      api: kubernetesAuthProvidersApiRef,
+      deps: {
+        gitlabAuthApi: gitlabAuthApiRef,
+        googleAuthApi: googleAuthApiRef,
+        microsoftAuthApi: microsoftAuthApiRef,
+        oktaAuthApi: oktaAuthApiRef,
+        oneloginAuthApi: oneloginAuthApiRef,
+      },
+      factory: ({
+        gitlabAuthApi,
+        googleAuthApi,
+        microsoftAuthApi,
+        oktaAuthApi,
+        oneloginAuthApi,
+      }) => {
+        const oidcProviders = {
+          gitlab: gitlabAuthApi,
+          google: googleAuthApi,
+          microsoft: microsoftAuthApi,
+          okta: oktaAuthApi,
+          onelogin: oneloginAuthApi,
+        };
+
+        return new KubernetesAuthProviders({
+          microsoftAuthApi,
+          googleAuthApi,
+          oidcProviders,
+        });
+      },
+    }),
+    createApiFactory({
```

Comment:
we need to also add `KubernetesProxyClient` to bring in the proxy api call to fetch Pod logs in both tekton and topology, can you add this KubernetesProxyClient in shared-react and in register it in both plugins?

```suggestion
    }),
  createApiFactory({
      api: kubernetesProxyApiRef,
      deps: {
        kubernetesApi: kubernetesApiRef,
      },
      factory: ({ kubernetesApi }) =>
        new KubernetesProxyClient({
          kubernetesApi,
        }),
    }),
    createApiFactory({
```